portfolio/src/main/webapp/script.js

```diff
@@ -44,3 +101,18 @@ function deletePost() {
 
   fetch(url, {method: 'POST'}).then(result => populatePosts());
 }
+
+/**
+  * Updates the score of a post in Datastore and on the DOM.
+  * @param {<button>} voteButton The HTML button this function will be assigned to.
+  * @param {VoteDirection} direction Specifies the change in 
+  * number of votes for a post when the button is clicked.
+  */
+function votePost(voteButton, direction) {
+  let score = direction == VoteDirection.UP ? voteButton.previousElementSibling : voteButton.nextElementSibling;
+  const comment = voteButton.parentElement.previousElementSibling.textContent;
```

Comment:
Sorry I actually didn't see that you had responded to my comment about this previously. The reason I don't like this line is that it's not very readable and it's also very sensitive to DOM changes with your comment layout. Here are some possible solutions

1. Add ```commentText``` as a parameter of this function, so you have access to it without having to navigate the DOM tree.
2. Add the card DIV as a parameter when creating this function, so you can use ```querySelector```.
3. Add an arbitrary ID to each card div, like ```comment1```, ```comment2```, etc, when creating the comments, so you can use ```querySelector```.

You actually have the same issue with finding score, but it's less serious.